services/gossip/adapter/peers.go

```diff
@@ -20,6 +25,14 @@ type gossipPeer struct {
 	hexOrbsAddress string
 }
 
+func NewGossipPeers(servicePeers []*services.GossipPeer) GossipPeers {
```

Comment:
its not very pretty that you are using your own `GossipPeer` and a type of the same name from the spec. now I also understand why you named the variable `servicePeers` which is a little cryptic (imho).

This will be more elegant if the local type was called `GossipEndpoint[Desc]` or something like that, then you can also change the param name to something simple like `peers` it makes a better separation of the packages imho..